mithril-common/src/entities/cardano_transactions_set_proof.rs

```diff
@@ -39,6 +41,38 @@ impl CardanoTransactionsSetProof {
 
         Ok(())
     }
+
+    #[cfg(feature = "test_tools")]
+    pub fn dummy() -> Self {
+        let transactions_hashes = vec![
+            "tx-1".to_string(),
+            "tx-2".to_string(),
+            "tx-3".to_string(),
+            "tx-4".to_string(),
+            "tx-5".to_string(),
+        ];
+        let leaves: Vec<MKTreeNode> = transactions_hashes
+            .iter()
+            .cloned()
+            .map(|h| h.into())
+            .collect();
+        let store = MKTreeStore::default();
+        let mktree = MKTree::new(&leaves, &store).unwrap();
+
+        Self {
+            transactions_hashes,
+            transactions_proof: mktree.compute_proof(&leaves).unwrap(),
+        }
+    }
+}
+
+impl From<CardanoTransactionsSetProof> for CardanoTransactionsSetProofMessagePart {
+    fn from(proof: CardanoTransactionsSetProof) -> Self {
+        Self {
+            transactions_hashes: proof.transactions_hashes,
+            proof: key_encode_hex(proof.transactions_proof).unwrap(),
```

Comment:
Maybe we can implement a `to_hex/from_hex` pair of associated functions to the `CardanoTransactionSetProof`, and implement `TryFrom` to avoid the unwrap?

Reply:
I added a `ProtocolMkProof` (aka a `ProtocolKey<MkProof>`) to use the `ProtocolKey::to_json_hex` method. Does that answer your point ?

Reply:
The `ProtocolKey` is a good idea :+1: 
Don't we need to get an error instead of a panic if the proof can't be encoded properly?